### PR TITLE
Add ostreecontainer support

### PR DESCRIPTION
**Pull request by kwozyman**
Backport of ostreecontainer command

Depends on:

 * [ ] https://github.com/pykickstart/pykickstart/pull/461
 * [ ] https://github.com/weldr/lorax/pull/1351

Resolves: https://issues.redhat.com/browse/RHEL-2250

### CODE REVIEW

**Review by VladimirSlavik — COMMENTED** [review]
Thanks! There's a lot of code, so lots of comments :)

Apart from the points below, there is an empty commit "Add new OSTree container source test (#2125655)" - what's the plan there?

File: Makefile.am [hunk 1]
```diff
@@ -78,7 +78,7 @@ TEMPLATE_RENDERER = scripts/jinja-render
 
 CONTAINER_ENGINE ?= podman
 # build/run tweaks for all containers, such as --cap-add
-CONTAINER_BUILD_ARGS ?= --no-cache
+CONTAINER_BUILD_ARGS ?= --no-cache --network=host
```
[the remark this answers is not in the document]
Comment (kwozyman) [reply Makefile.am]: I've remove this from the current PR

File: docs/release-notes/ostree-native-containers-support.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+:Type: Kickstart
+:Summary: Add support for OSTree native containers (#2125655)
```
[the remark this answers is not in the document]
Comment (kwozyman) [reply docs/release-notes/ostree-native-containers-support.rst]: I've removed this from the current PR

File: pyanaconda/core/constants.py [hunk 3]
```diff
@@ -429,11 +429,15 @@ class DisplayModes(Enum):
 # Types of the payload source.
 SOURCE_TYPE_LIVE_OS_IMAGE = "LIVE_OS_IMAGE"
 SOURCE_TYPE_LIVE_IMAGE = "LIVE_IMAGE"
+SOURCE_TYPE_LIVE_TAR = "LIVE_TAR"
```
[the remark this answers is not in the document]
Comment (kwozyman) [reply pyanaconda/core/constants.py]: I think you're right, I've removed all the references to SOURCE_TYPE_LIVE_TAR

File: pyanaconda/core/kickstart/commands.py [hunk 4]
```diff
@@ -57,6 +57,12 @@
 from pykickstart.commands.ostreesetup import RHEL9_OSTreeSetup as OSTreeSetup
 from pykickstart.commands.partition import RHEL9_Partition as Partition
 from pykickstart.commands.raid import RHEL9_Raid as Raid
+from pykickstart.commands.ostreecontainer import F38_OSTreeContainer as OSTreeContainer
+from pykickstart.commands.ostreesetup import F38_OSTreeSetup as OSTreeSetup
+from pykickstart.commands.ostreecontainer import F38_OSTreeContainer as OSTreeContainer
+from pykickstart.commands.ostreesetup import F21_OSTreeSetup as OSTreeSetup
+from pykickstart.commands.partition import F34_Partition as Partition
+from pykickstart.commands.raid import F29_Raid as Raid
```
[the remark this answers is not in the document]
Comment (kwozyman) [reply pyanaconda/core/kickstart/commands.py]: Correct, I've updated this area to only reference RHEL9_* to match PyKickstart

**Comment by rvykydal** [comment]
I should have mentioned it here explicitly (I noted it only in the RHEL jira issue) I am working on alternative PR https://github.com/rhinstaller/anaconda/pull/5399.

**Comment by jkonecny12** [comment]
Closing this PR on behalf of https://github.com/rhinstaller/anaconda/pull/5399 which is based on this PR. Thanks a lot for your work @kwozyman. 
The https://github.com/rhinstaller/anaconda/pull/5399 is ready for next RHEL-9 release.

**Pull request closed without merging**